retrievalmarket/impl/integration_test.go

```diff
@@ -508,9 +508,7 @@ CurrentInterval: %d
 				t.FailNow()
 			case clientDealState = <-clientDealStateChan:
 			}
-			if testCase.failsUnseal {
-				assert.Equal(t, retrievalmarket.DealStatusErrored, clientDealState.Status)
-			} else if testCase.cancelled {
+			if testCase.failsUnseal || testCase.cancelled {
```

Comment:
If there's an error unsealing on the provider, the best we can hope for on the client side is to receive a cancel message.
This test only passed before because the statemachine on the client was already shutdown when it received the Cancel event, so firing the event caused an error.
I fixed that bug so that the statemachine just ignores a Cancel event when the statemachine has already been shutdown